tests/tui/test_spec_file_load.py

```diff
@@ -0,0 +1,220 @@
+# -*- coding: utf-8 -*-
+
+from asyncio import sleep
+from functools import partial
+from pathlib import Path
+
+from .pilot_functions import (
+    _set_work_dir,
+    check_and_run_tab_refresh,
+    settable_scroll_screen_down,
+)
+
+widget_label_map = {
+    "dualReg_1": "#feature_item_4_flabel",
+    "corrMatrix_1": "#feature_item_1_flabel",
+    "taskBased_1": "#feature_item_0_flabel",
+    "fALFF_1": "#feature_item_2_flabel",
+    "seedCorr_1": "#feature_item_3_flabel",
+    "reHo_1": "#feature_item_5_flabel",
+    "taskBased_2": "#feature_item_6_flabel",
+    "fALFF_2": "#feature_item_7_flabel",
+    "preproc_1": "#feature_item_8_flabel",
+}
+
+
+async def run_before(
+    pilot, data_path=None, spec_file_dir_path=None, feature_label=None, scroll_to_remaining_part=False
+) -> None:
+    # always reload the app first, there is some strange crossinteraction between tests, nothing else helped except using
+    # -n 2 flag for the pytest, i.e., running each test with a separate worker
+    feature_label = feature_label if feature_label is not None else None
+
+    pilot.app.reload_ui()
+    if isinstance(data_path, Path):
+        data_path = str(data_path)
+    if isinstance(spec_file_dir_path, Path):
+        work_dir_path = str(spec_file_dir_path)
+        # work_dir_path = "/makethisfail/"
+
+    await _set_work_dir(pilot, work_dir_path, load_from_spec_file=True)
+    # For some reason the button remains focussed when I do it locally, but it is not focussed when it runs through CI,
+    # the tab should prevent this.
+    if feature_label is None:
+        await check_and_run_tab_refresh(pilot)
+        await settable_scroll_screen_down(pilot, 50)
+    else:
+        await sleep(10)
+        await pilot.press("f")
+        if scroll_to_remaining_part:
+            # await pilot.click(offset=(15, feature_number))
+            await pilot.click("#sidebar")
```

Comment:
Thanks for starting with these changes! How long do you think it will take to update all of the interactions? Is it realistic for this pull request?

Reply:
i think it is now all done with the ids